app/uk/gov/hmrc/individualsdetailsapi/connectors/IntegrationFrameworkConnector.scala

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.individualsdetailsapi.connectors
+
+import javax.inject.Inject
+import org.joda.time.{Interval, LocalDate}
+import play.api.Logger
+import uk.gov.hmrc.domain.Nino
+import uk.gov.hmrc.http.logging.Authorization
+import uk.gov.hmrc.http.{
+  HeaderCarrier,
+  HttpClient,
+  NotFoundException,
+  TooManyRequestException,
+  Upstream4xxResponse
+}
+import uk.gov.hmrc.individualsdetailsapi.domains.integrationframework.{
+  IFDetails,
+  IFDetailsResponse
+}
+import uk.gov.hmrc.play.bootstrap.config.ServicesConfig
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class IntegrationFrameworkConnector @Inject()(
```

Comment:
I have renamed this to `IfConnector` in my implementation to keep it short. Might be good to do the same in the others?